sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -256,10 +256,23 @@ public static bool IsStringEmpty(this ExpressionSyntax expression, SemanticModel
         }
 
         public static bool IsMethodInvocation(this InvocationExpressionSyntax expression, KnownType type, string methodName, SemanticModel semanticModel) =>
+            expression.NameIs(methodName) &&
             semanticModel.GetSymbolInfo(expression).Symbol is IMethodSymbol methodSymbol &&
             methodSymbol.IsInType(type) &&
             methodName.Equals(methodSymbol.Name, StringComparison.InvariantCulture);
 
+        public static bool IsMethodInvocation(this InvocationExpressionSyntax expression, ImmutableArray<KnownType> types, string methodName, SemanticModel semanticModel) =>
+            expression.NameIs(methodName) &&
+            semanticModel.GetSymbolInfo(expression).Symbol is IMethodSymbol methodSymbol &&
+            methodSymbol.IsInType(types) &&
+            methodName.Equals(methodSymbol.Name, StringComparison.InvariantCulture);
+
+        public static bool IsPropertyInvocation(this MemberAccessExpressionSyntax expression, ImmutableArray<KnownType> types, string propertyName, SemanticModel semanticModel) =>
+            expression.NameIs(propertyName) &&
+            semanticModel.GetSymbolInfo(expression).Symbol is IPropertySymbol propertySymbol &&
+            propertySymbol.IsInType(types) &&
+            propertyName.Equals(propertySymbol.Name, StringComparison.InvariantCulture);

```

Comment:
like the above comment, the latest `Equals` seems to be useless.

Reply:
damn, it seems I missed this one. I'll clean up soon